tests/lib/rules/require-expect.js

```diff
@@ -0,0 +1,195 @@
+            }],
+            options: ["except-simple"]
+        },
+
+        // `expect` does not count when used inside of a block.
+        {
+            code: "test('name', function(assert) { function name() { assert.expect(0); assert.ok(true) } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // assert in outer test context and nested in a block
+        {
+            code: "test('name', function(assert) { assert.ok(true); if (true) { assert.ok(true); } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // Deeply nested
+        {
+            code: "test('name', function(assert) { if (true) { if (true) { assert.ok(true); } } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // Sending assert to a function
+        {
+            code: [
+                "function myAssertion(a, assert, c) { assert.ok(true); }",
+                "test('name', function(assert) { myAssertion(null, assert, null); });"
+            ].join(returnAndIndent),
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // nested modules
+        {
+            code: [
+                "module('a', function() {",
+                "    test('a - test', function(assert) {",
+                "        if (false) {",
+                "            assert.ok(true, 'needs expect');",
+                "        }",
+                "    });",
+                "",
+                "    module('b', function() {",
+                "        test('b - test', function(assert) {",
+                "            assert.expect(1);",
+                "            assert.ok(true, 'has expect');",
+                "        });",
+                "    });",
+                "});"
+            ].join(returnAndIndent),
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // "always" confugration - simple case
+        {
+            code: "test('name', function(assert) { assert.ok(true) })",
+            options: [],
+            errors: [{ message: EXCEPT_SIMPLE_ERROR_MESSAGE }]
+        },
+
+        // "always" configuration checking that "expect" is called on assert.
+        {
+            code: "test('name', function(assert) { other.expect(1); assert.ok(true); });",
+            options: [],
+            errors: [{ message: EXCEPT_SIMPLE_ERROR_MESSAGE }]
+        }
```

Comment:
"always" strategy but EXCEPT_SIMPLE_ERROR_MESSAGE, does this need to be re-examined?